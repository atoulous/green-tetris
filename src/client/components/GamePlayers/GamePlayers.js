import React from 'react';
import PropTypes from 'prop-types';

import RaisedButton from 'material-ui/RaisedButton';
import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';


import './GamePlayers.scss';


const GamePlayers = ({ game, dispatch }) => {
  const handleKick = player => () => {
    console.log('KICK THE PLAYER:', player);
    /*
    ** Change state here with dispatch
    */
  };

  return (
    <div className="game-players">
      <Table selectable={false}>
        <TableHeader adjustForCheckbox={false} displaySelectAll={false}>
          <TableRow>
            <TableHeaderColumn>Name</TableHeaderColumn>
            <TableHeaderColumn>Is Ready ?</TableHeaderColumn>
            <TableHeaderColumn />
          </TableRow>
        </TableHeader>
        <TableBody displayRowCheckbox={false}>
          {
            game.currentPlayers.map(player => (
              <TableRow key={player.id}>
                <TableRowColumn>{player.name}</TableRowColumn>
                <TableRowColumn>{player.isReady}</TableRowColumn>
                <TableRowColumn>
                  <RaisedButton label="Kick" onClick={handleKick(player)} />
                </TableRowColumn>
              </TableRow>
                ))
        }
        </TableBody>
      </Table>
    </div>
  );
};

GamePlayers.propTypes = {
  game: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default GamePlayers;
