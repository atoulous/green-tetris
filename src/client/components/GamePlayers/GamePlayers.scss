.game-players {
  flex: 1;

  td {
    vertical-align: middle;
  }
}
