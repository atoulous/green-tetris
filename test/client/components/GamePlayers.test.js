import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import GamePlayers from '../../../src/client/components/GamePlayers/GamePlayers';

it('', () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <GamePlayers
          dispatch={() => false}
          player={{ id: 'oijoijoij', isReady: true }}
          game={{
        id: 'oijoijoij',
        hasStarted: false,
        masterId: 'oijoijoij',
        speed: { label: 'Normal' },
        size: { label: 'Normal' },
        maxPlayers: 3,
        players: [
          { nickname: 'thibault1', id: '1', isReady: true },
          { nickname: 'thibault2', id: '2', isReady: true },
          { nickname: 'thibault3', id: '3', isReady: true },
          { nickname: 'thibault4', id: '4', isReady: true },
        ]
      }}
        />
      </MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

