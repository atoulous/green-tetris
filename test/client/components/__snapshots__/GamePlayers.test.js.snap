// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="game-players"
>
  <div
    style={
      Object {
        "height": "auto",
        "muiPrepared": true,
        "overflow": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "muiPrepared": true,
        }
      }
    >
      <table
        className={undefined}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderCollapse": "collapse",
            "borderSpacing": 0,
            "fontFamily": "Roboto, sans-serif",
            "tableLayout": "fixed",
            "width": "100%",
          }
        }
      >
        <thead
          className={undefined}
          style={
            Object {
              "borderBottom": "1px solid #e0e0e0",
              "muiPrepared": true,
            }
          }
        >
          <tr
            className={undefined}
            style={
              Object {
                "borderBottom": "1px solid #e0e0e0",
                "color": "rgba(0, 0, 0, 0.87)",
                "height": 48,
                "muiPrepared": true,
              }
            }
          >
            <th
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "color": "#9e9e9e",
                  "fontSize": 12,
                  "fontWeight": "normal",
                  "height": 56,
                  "muiPrepared": true,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "position": "relative",
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              Name
            </th>
            <th
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "color": "#9e9e9e",
                  "fontSize": 12,
                  "fontWeight": "normal",
                  "height": 56,
                  "muiPrepared": true,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "position": "relative",
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              Is Ready ?
            </th>
            <th
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "color": "#9e9e9e",
                  "fontSize": 12,
                  "fontWeight": "normal",
                  "height": 56,
                  "muiPrepared": true,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "position": "relative",
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            />
          </tr>
        </thead>
      </table>
    </div>
    <div
      style={
        Object {
          "height": "inherit",
          "muiPrepared": true,
          "overflowX": "hidden",
          "overflowY": "auto",
        }
      }
    >
      <table
        className={undefined}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderCollapse": "collapse",
            "borderSpacing": 0,
            "fontFamily": "Roboto, sans-serif",
            "tableLayout": "fixed",
            "width": "100%",
          }
        }
      >
        <tbody
          style={
            Object {
              "muiPrepared": true,
            }
          }
        >
          <tr
            className={undefined}
            style={
              Object {
                "borderBottom": "1px solid #e0e0e0",
                "color": "rgba(0, 0, 0, 0.87)",
                "height": 48,
                "muiPrepared": true,
              }
            }
          >
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              thibault1
            </td>
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              <div
                className={undefined}
                disabled={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "backgroundColor": "#74d680",
                    "border": 10,
                    "borderRadius": 16,
                    "boxShadow": null,
                    "boxSizing": "border-box",
                    "cursor": "default",
                    "display": "flex",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": "inherit",
                    "fontWeight": "inherit",
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "textDecoration": "none",
                    "verticalAlign": null,
                    "whiteSpace": "nowrap",
                    "width": "fit-content",
                  }
                }
                tabIndex={0}
              >
                <span
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontSize": 14,
                      "fontWeight": 400,
                      "lineHeight": "32px",
                      "muiPrepared": true,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "userSelect": "none",
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  Ready
                </span>
              </div>
            </td>
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              <div
                className={undefined}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "backgroundColor": "#ffffff",
                    "borderRadius": 2,
                    "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "minWidth": 88,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <button
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                      "backgroundColor": "#ffffff",
                      "border": 10,
                      "borderRadius": 2,
                      "boxSizing": "border-box",
                      "cursor": "pointer",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": "inherit",
                      "fontWeight": "inherit",
                      "height": 36,
                      "lineHeight": "36px",
                      "margin": 0,
                      "muiPrepared": true,
                      "outline": "none",
                      "padding": 0,
                      "position": "relative",
                      "textAlign": "center",
                      "textDecoration": "none",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "verticalAlign": null,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                  >
                    <div
                      style={
                        Object {
                          "backgroundColor": false,
                          "borderRadius": 2,
                          "height": 36,
                          "muiPrepared": true,
                          "top": 0,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        }
                      }
                    >
                      <span
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                            "fontSize": 14,
                            "fontWeight": 500,
                            "letterSpacing": 0,
                            "margin": 0,
                            "muiPrepared": true,
                            "opacity": 1,
                            "paddingLeft": 16,
                            "paddingRight": 16,
                            "position": "relative",
                            "textTransform": "uppercase",
                            "userSelect": "none",
                          }
                        }
                      >
                        Kick
                      </span>
                    </div>
                  </div>
                </button>
              </div>
            </td>
          </tr>
          <tr
            className={undefined}
            style={
              Object {
                "borderBottom": "1px solid #e0e0e0",
                "color": "rgba(0, 0, 0, 0.87)",
                "height": 48,
                "muiPrepared": true,
              }
            }
          >
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              thibault2
            </td>
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              <div
                className={undefined}
                disabled={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "backgroundColor": "#74d680",
                    "border": 10,
                    "borderRadius": 16,
                    "boxShadow": null,
                    "boxSizing": "border-box",
                    "cursor": "default",
                    "display": "flex",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": "inherit",
                    "fontWeight": "inherit",
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "textDecoration": "none",
                    "verticalAlign": null,
                    "whiteSpace": "nowrap",
                    "width": "fit-content",
                  }
                }
                tabIndex={0}
              >
                <span
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontSize": 14,
                      "fontWeight": 400,
                      "lineHeight": "32px",
                      "muiPrepared": true,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "userSelect": "none",
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  Ready
                </span>
              </div>
            </td>
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              <div
                className={undefined}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "backgroundColor": "#ffffff",
                    "borderRadius": 2,
                    "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "minWidth": 88,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <button
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                      "backgroundColor": "#ffffff",
                      "border": 10,
                      "borderRadius": 2,
                      "boxSizing": "border-box",
                      "cursor": "pointer",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": "inherit",
                      "fontWeight": "inherit",
                      "height": 36,
                      "lineHeight": "36px",
                      "margin": 0,
                      "muiPrepared": true,
                      "outline": "none",
                      "padding": 0,
                      "position": "relative",
                      "textAlign": "center",
                      "textDecoration": "none",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "verticalAlign": null,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                  >
                    <div
                      style={
                        Object {
                          "backgroundColor": false,
                          "borderRadius": 2,
                          "height": 36,
                          "muiPrepared": true,
                          "top": 0,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        }
                      }
                    >
                      <span
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                            "fontSize": 14,
                            "fontWeight": 500,
                            "letterSpacing": 0,
                            "margin": 0,
                            "muiPrepared": true,
                            "opacity": 1,
                            "paddingLeft": 16,
                            "paddingRight": 16,
                            "position": "relative",
                            "textTransform": "uppercase",
                            "userSelect": "none",
                          }
                        }
                      >
                        Kick
                      </span>
                    </div>
                  </div>
                </button>
              </div>
            </td>
          </tr>
          <tr
            className={undefined}
            style={
              Object {
                "borderBottom": false,
                "color": "rgba(0, 0, 0, 0.87)",
                "height": 48,
                "muiPrepared": true,
              }
            }
          >
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              thibault3
            </td>
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              <div
                className={undefined}
                disabled={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "backgroundColor": "#74d680",
                    "border": 10,
                    "borderRadius": 16,
                    "boxShadow": null,
                    "boxSizing": "border-box",
                    "cursor": "default",
                    "display": "flex",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": "inherit",
                    "fontWeight": "inherit",
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "textDecoration": "none",
                    "verticalAlign": null,
                    "whiteSpace": "nowrap",
                    "width": "fit-content",
                  }
                }
                tabIndex={0}
              >
                <span
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontSize": 14,
                      "fontWeight": 400,
                      "lineHeight": "32px",
                      "muiPrepared": true,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "userSelect": "none",
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  Ready
                </span>
              </div>
            </td>
            <td
              className={undefined}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "inherit",
                  "fontSize": 13,
                  "height": 48,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "textAlign": "left",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                }
              }
            >
              <div
                className={undefined}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "backgroundColor": "#ffffff",
                    "borderRadius": 2,
                    "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "minWidth": 88,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <button
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                      "backgroundColor": "#ffffff",
                      "border": 10,
                      "borderRadius": 2,
                      "boxSizing": "border-box",
                      "cursor": "pointer",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": "inherit",
                      "fontWeight": "inherit",
                      "height": 36,
                      "lineHeight": "36px",
                      "margin": 0,
                      "muiPrepared": true,
                      "outline": "none",
                      "padding": 0,
                      "position": "relative",
                      "textAlign": "center",
                      "textDecoration": "none",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "verticalAlign": null,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                  >
                    <div
                      style={
                        Object {
                          "backgroundColor": false,
                          "borderRadius": 2,
                          "height": 36,
                          "muiPrepared": true,
                          "top": 0,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        }
                      }
                    >
                      <span
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.87)",
                            "fontSize": 14,
                            "fontWeight": 500,
                            "letterSpacing": 0,
                            "margin": 0,
                            "muiPrepared": true,
                            "opacity": 1,
                            "paddingLeft": 16,
                            "paddingRight": 16,
                            "position": "relative",
                            "textTransform": "uppercase",
                            "userSelect": "none",
                          }
                        }
                      >
                        Kick
                      </span>
                    </div>
                  </div>
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
